packages/shared/src/contexts/AuthContext.tsx

```diff
@@ -87,11 +88,16 @@ export const AuthContextProvider = ({
   refetchBoot,
   visit,
   isLegacyLogout,
+  firstLoad,
 }: AuthContextProviderProps): ReactElement => {
   const [loginState, setLoginState] = useState<LoginState | null>(null);
   const endUser = user && 'providers' in user ? user : null;
   const referral = user?.referrer;
 
+  if (firstLoad === true && endUser && !endUser?.infoConfirmed) {
```

Comment:
Shouldn't this be inside a `useEffect`?

Reply:
Don't think it has to, it's all passed down which will cause re-evaluation each time.
All my test cases seem to work, what would be the positive of going useEffect for this?

Reply:
That it will happen only once

Reply:
I think that would actually give us a lot of headaches and complicates as it's always delayed then.
Worse case this would call twice, but because of the state also can't happen right? (The firstLoad state)

Reply:
Ok, let's do it :)